### DISCUSSION TITLE
Directly reading a passworded HLS stream via web browser?

### DISCUSSION THREAD

**Discussion by legalmentemama** [bluenviron/mediamtx#2433] [Questions]
How can I pass the user and password in an iframe or HTTP GET request to connect to an HLS stream that's protected by user and password?

Say I have

192.168.1.2:5152/myvideopath/

My iframe link is literally just that for the embed, and it pops-up a normal authorization window for user and password. Is there a way to directly pass these in the link? As in, to have a small landing site where users can specify a video path, user and password to watch a stream.

**Comment by aler9** [comment]
Hello, passing HTTP credentials to iframes is blocked by all browsers for security reasons. In my opinion, the best way to create private streams is to embed some sort of security token into the path name, for instance:

```
192.168.1.2:5152/myvideopath_h2rtg98hw234ry58h
```

[a comment by an automated account is not shown]
